packages/analytics-browser/src/browser-client.ts

```diff
@@ -263,14 +272,21 @@ export class AmplitudeBrowser extends AmplitudeCore implements BrowserClient {
   async process(event: Event) {
     const currentTime = Date.now();
     const isEventInNewSession = isNewSession(this.config.sessionTimeout, this.config.lastEventTime);
-
+    const shouldTrackNewCampaign = this.webAttribution && (await this.webAttribution.shouldTrackNewCampaign());
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const shouldSetSessionId = shouldTrackNewCampaign && this.webAttribution!.options.resetSessionOnNewCampaign;
     if (
       event.event_type !== DEFAULT_SESSION_START_EVENT &&
       event.event_type !== DEFAULT_SESSION_END_EVENT &&
-      (!event.session_id || event.session_id === this.getSessionId()) &&
-      isEventInNewSession
+      (!event.session_id || event.session_id === this.getSessionId())
     ) {
-      this.setSessionId(currentTime);
+      if (isEventInNewSession || shouldSetSessionId) {
+        await this.setSessionId(currentTime, shouldTrackNewCampaign);
+      } else if (!isEventInNewSession && shouldTrackNewCampaign) {
+        // web attribution should be track during the middle of the session if there has any new campaign
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        await this.webAttribution!.track().promise;
```

Comment:
Make this as required(!) instead optional (?) because 
1. shouldTrackNewCampaign already make sure this.webAttribution exist
2. the test coverage needs to cover `?` 
Will revert this back to optional and try other ways to bypass the test coverage.